src/components/chartRepo/ChartRepo.tsx

```diff
@@ -261,8 +320,68 @@ function ChartForm({ id = null, name = "", active = false, url = "", authMode =
             />
 
             <div className="form__row form__row--two-third">
-                <CustomInput autoComplete="off" value={state.name.value} onChange={handleOnChange} name="name" error={state.name.error} label="Name*" />
-                <CustomInput autoComplete="off" value={state.url.value} onChange={handleOnChange} name="url" error={state.url.error} label="URL*" />
+                <CustomInput
+                    autoComplete="off"
+                    value={state.name.value}
+                    onChange={handleOnChange}
+                    name="name"
+                    error={state.name.error}
+                    label="Name*"
+                />
+                <CustomInput
+                    autoComplete="off"
+                    value={state.url.value}
+                    onChange={handleOnChange}
+                    name="url"
+                    error={state.url.error}
+                    label="URL*"
+                />
+                {chartRepoType === 'PUBLIC' ? (
+                    <></>
+                ) : (
+                    <>
+                        <CustomInput
+                            autoComplete="off"
+                            value={customState.username.value}
+                            onChange={customHandleChange}
+                            name="username"
+                            error={customState.username.error}
+                            label="Username*"
+                            labelClassName="mt-12"
+                        />
+                        <ProtectedInput
+                            value={customState.password.value}
+                            onChange={customHandleChange}
+                            name="password"
+                            error={customState.password.error}
+                            label="Password*"
+                            labelClassName="mt-12"
+                        />
+                    </>
+                )}
+                {id && authMode === 'USERNAME_PASSWORD' ? (
+                    <>
+                        <CustomInput
```

Comment:
Repetitive code. Please check & update as required.

Reply:
and use `&&` syntax

Reply:
done